pinot-broker/src/main/java/com/linkedin/pinot/broker/queryquota/TableQueryQuotaManager.java

```diff
@@ -0,0 +1,265 @@
+    }
+    // Log a warning if MaxQueriesPerSecond are set different.
+    if ((offlineQuotaConfig != null && !Strings.isNullOrEmpty(offlineQuotaConfig.getMaxQueriesPerSecond()))
+        && (realtimeQuotaConfig != null && !Strings.isNullOrEmpty(realtimeQuotaConfig.getMaxQueriesPerSecond()))) {
+      if (!offlineQuotaConfig.getMaxQueriesPerSecond().equals(realtimeQuotaConfig.getMaxQueriesPerSecond())) {
+        LOGGER.warn("Attention! The values of MaxQueriesPerSecond for table {} are set different! Offline table qps quota: {}, Real-time table qps quota: {}",
+            rawTableName, offlineQuotaConfig.getMaxQueriesPerSecond(), realtimeQuotaConfig.getMaxQueriesPerSecond());
+      }
+    }
+
+    // Create rate limiter
+    createRateLimiter(tableName, brokerResource, tableConfig.getQuotaConfig());
+  }
+
+
+  /**
+   * Drop table query quota.
+   * @param tableConfig table config.
+   * @param brokerResource broker resource which stores all the broker states of each table.
+   * */
+  public void dropTableQueryQuota(TableConfig tableConfig, ExternalView brokerResource) {
+    String tableName = tableConfig.getTableName();
+    String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+    LOGGER.info("Dropping rate limiter for table {}", rawTableName);
+
+    removeRateLimiter(brokerResource, tableName, tableConfig.getQuotaConfig());
+  }
+
+  /** Remove or update rate limiter if another table with the same raw table name but different type is still using the quota config.
+   * @param brokerResource broker resource which stores all the broker states of each table.
+   * @param tableName original table name
+   * @param quotaConfig old quota config
+   * */
+  private void removeRateLimiter(ExternalView brokerResource, String tableName, QuotaConfig quotaConfig) {
+    if (quotaConfig == null || Strings.isNullOrEmpty(quotaConfig.getMaxQueriesPerSecond()) || !_rateLimiterMap.containsKey(tableName)) {
+      // Do nothing if the current table doesn't specify qps quota.
+      LOGGER.info("No qps quota is specified for table: {}", tableName);
+      return;
+    }
+    if (brokerResource == null) {
+      LOGGER.warn("Failed to drop qps quota for table {}. No broker resource connected!", tableName);
+      return;
+    }
+    _rateLimiterMap.remove(tableName);
+  }
+
+  /**
+   * Get QuotaConfig from property store.
+   * @param rawTableName table name without table type.
+   * @param tableType table type: offline or real-time.
+   * @return QuotaConfig, which could be null.
+   * */
+  private QuotaConfig getQuotaConfigFromPropertyStore(String rawTableName, CommonConstants.Helix.TableType tableType) {
+    ZkHelixPropertyStore<ZNRecord> propertyStore = _helixManager.getHelixPropertyStore();
+
+    String tableNameWithType = TableNameBuilder.forType(tableType).tableNameWithType(rawTableName);
+    TableConfig tableConfig = ZKMetadataProvider.getTableConfig(propertyStore, tableNameWithType);
+    if (tableConfig == null) {
```

Comment:
Should log a warning if we can't fetch the table configuration, this really shouldn't happen.

Reply:
This method is used to fetch the table config with another table type. E.g. if the the table type is offline, then we try to fetch the real-time table config, and wise versa. And it's possible that there's only one table config in the property store.